src/transformers/models/auto/image_processing_auto.py

```diff
@@ -413,7 +416,37 @@ def from_pretrained(cls, pretrained_model_name_or_path, *inputs, **kwargs):
         trust_remote_code = kwargs.pop("trust_remote_code", None)
         kwargs["_from_auto"] = True
 
-        config_dict, _ = ImageProcessingMixin.get_image_processor_dict(pretrained_model_name_or_path, **kwargs)
+        # Resolve the image processor config filename
+        if "image_processor_filename" in kwargs:
+            image_processor_filename = kwargs.pop("image_processor_filename")
+        elif is_timm_local_checkpoint(pretrained_model_name_or_path):
+            image_processor_filename = CONFIG_NAME
+        else:
+            image_processor_filename = IMAGE_PROCESSOR_NAME
+
+        # Load the image processor config
+        try:
+            # Main path for all transformers models and local TimmWrapper checkpoints
+            config_dict, _ = ImageProcessingMixin.get_image_processor_dict(
+                pretrained_model_name_or_path, image_processor_filename=image_processor_filename, **kwargs
+            )
+        except Exception as initial_exception:
+            # Fallback path for Hub TimmWrapper checkpoints. Timm models' image processing is saved in `config.json`
+            # instead of `preprocessor_config.json`. Because this is an Auto class and we don't have any information
+            # except the model name, the only way to check if a remote checkpoint is a timm model is to try to
+            # load `config.json` and if it fails with some error, we raise the initial exception.
+            try:
+                config_dict, _ = ImageProcessingMixin.get_image_processor_dict(
+                    pretrained_model_name_or_path, image_processor_filename=CONFIG_NAME, **kwargs
+                )
+            except Exception:
+                raise initial_exception
+
```

Comment:
A remark here - that's true for any processor with a `preprocessor_config.json`, but would it make sense to sanitize the inputs a bit? On community checkpoints, there's extra keys that are unused, for instance `model_args` that contains duplicated information. 
(I'm asking because we're already using a try/catch pattern here so that allows some branching)

Reply:
basically I I load https://huggingface.co/prov-gigapath/prov-gigapath/blob/main/config.json, I'll end up with an `ImageProcessor` object that has keys I can't make much use of in transformers, such as `model_args`, so I wondered if it made sense to filter the contents of that `config.json` to an expected schema

Reply:
Ok, it makes sense now. I made it on the image-processor init level, see https://github.com/huggingface/transformers/pull/34564/commits/fd7b646f3789413c75020fe2b7c6735c0dc8e9c4

Is it what you had in mind?

```python
>>> from transformers import AutoImageProcessor, AutoConfig

>>> image_processor = AutoImageProcessor.from_pretrained("prov-gigapath/prov-gigapath")
>>> print(image_processor)

TimmWrapperImageProcessor {
  "architecture": "vit_giant_patch14_dinov2",
  "data_config": {
    "crop_mode": "center",
    "crop_pct": 1.0,
    "input_size": [
      3,
      224,
      224
    ],
    "interpolation": "bicubic",
    "mean": [
      0.485,
      0.456,
      0.406
    ],
    "std": [
      0.229,
      0.224,
      0.225
    ]
  },
  "image_processor_type": "TimmWrapperImageProcessor"
}
```

Reply:
yes, exactly! that looks good